spec/requests/conversion_hosts_spec.rb

```diff
@@ -70,45 +76,60 @@
 
     let(:expected_attributes) { %w(id name resource_type resource_id version) }
 
+    before do
+      allow_any_instance_of(ConversionHost).to receive(:enable_conversion_host_role).and_return(true)
+    end
+
     it "supports single conversion host creation" do
       api_basic_authorize(collection_action_identifier(:conversion_hosts, :create))
 
       post(api_conversion_hosts_url, :params => sample_conversion_host_from_vm)
 
       expect(response).to have_http_status(:ok)
-      expect_result_resources_to_include_keys("results", expected_attributes)
 
-      conversion_host_id = response.parsed_body["results"].first["id"]
+      results = response.parsed_body["results"].first
 
-      expect(ConversionHost.find_by(:resource_id => vm.id).id).to eql(conversion_host_id.to_i)
+      expect(results['success']).to be_truthy
+      expect(results['href']).to eql('http://www.example.com/api/conversion_hosts/')
+      expect(results['message']).to eql("Enabling resource id:#{vm.id} type:Vm")
     end
 
-    it "supports single conversion host creation via action" do
+    it "supports multiple conversion host creation" do
       api_basic_authorize(collection_action_identifier(:conversion_hosts, :create))
 
-      post(api_conversion_hosts_url, :params => gen_request(:create, sample_conversion_host_from_vm))
+      conversion_hosts = [sample_conversion_host_from_vm, sample_conversion_host_from_host]
+      post(api_conversion_hosts_url, :params => gen_request(:create, conversion_hosts))
 
       expect(response).to have_http_status(:ok)
-      expect_result_resources_to_include_keys("results", expected_attributes)
 
-      conversion_host_id = response.parsed_body["results"].first["id"]
-      expect(ConversionHost.find_by(:resource_id => vm.id).id).to eql(conversion_host_id.to_i)
+      results = response.parsed_body["results"]
+
+      expect(results).to match_array([
+        a_hash_including("message" => "Enabling resource id:#{vm.id} type:Vm"),
+        a_hash_including("message" => "Enabling resource id:#{host.id} type:Host"),
+      ])
     end
+  end
 
-    it "supports multiple conversion host creation" do
-      api_basic_authorize(collection_action_identifier(:conversion_hosts, :create))
+  context "disable" do
```

Comment:
how about the existing `context "delete" do"`?

Reply:
Ok, done.